fs/base.py

```diff
@@ -310,6 +312,22 @@ def setinfo(self, path, info):
 
         """
 
+    @staticmethod
+    def quote(path_snippet):
```

Comment:
@willmcgugan Is `base.py` the "right" place for this new `quote` function?
@chfw I wonder if `quote` is too generic a name and something more descriptive would be better? :man_shrugging:

Reply:
@lurch I could find a better place for it. I am happy to put it elsewhere.

url_quote?

Reply:
I think we should move it out of the base class, wouldn't want to imply it was part of the core interface. Feel free to add a utility file of some sort. Maybe _url_tools.py

@chfw Would you mind adding a docstring and typing comments for this function.

Reply:
added